tests/Tests/500__jupyterhub/test-jupyterlab-git.robot

```diff
@@ -0,0 +1,176 @@
+    ${commit_message}=    Catenate    ${COMMIT_MSG}    ${randnum}
+    Push Some Changes to Repo
+    ...    ${GITHUB_USER.USERNAME}
+    ...    ${GITHUB_USER.TOKEN}
+    ...    ${FILE_PATH}
+    ...    ${REPO_URL}
+    ...    ${commit_message}
+    [Teardown]    Clean Up Server
+
+Verify Updating Project With Changes From Git Repository
+    [Documentation]    Verifies that changes has been pulled successfully to local repository
+    [Tags]    ODS-324
+    ...       Sanity    Tier1
+    Set Staging Status
+    Clone Git Repository And Open    ${REPO_URL}    ${FILE_PATH}
+    Sleep    1s
+    Open New Notebook
+    ${commit_msg1}=    Get Last Commit Message
+    Add and Run JupyterLab Code Cell in Active Notebook    ! mkdir ../folder/
+    Sleep    2s
+    Open Folder or File    folder
+
+    ${randnum}=    Generate Random String    9    [NUMBERS]
+    ${commit_message}=    Catenate    ${COMMIT_MSG}    ${randnum}
+    #now do here some changes
+    Push Some Changes to Repo
+    ...    ${GITHUB_USER.USERNAME}
+    ...    ${GITHUB_USER.TOKEN}
+    ...    folder/${FILE_PATH}
+    ...    ${REPO_URL}
+    ...    ${commit_message}
+    Close All JupyterLab Tabs
+    Open Folder or File    ${DIR_NAME}
+    Open With JupyterLab Menu    Git    Pull from Remote
+    Sleep    2s
+    Open New Notebook
+    ${commit_msg2}=    Get Last Commit Message
+    Should Not Be Equal    ${commit_msg2}    ${commit_msg1}
+    [Teardown]    Clean Up Server
+
+
+*** Keywords ***
+Server Setup
+    [Documentation]    Suite Setup
+    Begin Web Test
+    Launch JupyterHub Spawner From Dashboard
+    Spawn Notebook With Arguments    image=s2i-minimal-notebook    size=Default
+
+Push Some Changes To Repo
+    [Documentation]    Make some changes in ${filepath} and push to remote repo
+    [Arguments]    ${github username}    ${token}    ${filepath}    ${githublink}    ${commitmsgg}
+    Clone Git Repository In Current Folder    ${githublink}
+    Close All JupyterLab Tabs
+    Open Folder or File    ${filepath}
+    Enter Text In File And Save    code=print("Hi Hello")
+    Set Staging Status    status=ON
+    Commit Changes    commit_message=${commitmsgg}    name=${GITHUB_USER.USERNAME}    email_id=${GITHUB_USER.EMAIL}
+    Push Changes To Remote    github_username=${GITHUB_USER.USERNAME}    token=${GITHUB_USER.TOKEN}
+    Set Staging Status    status=OFF
+    Close All JupyterLab Tabs
+    Sleep    2s
+    Open New Notebook
+    ${output}=    Get Last Commit Message
+    Should Be Equal    ${commitmsgg.strip()}    ${output.strip()}
+
+Open Folder Or File
+    [Documentation]    Opens the folder or file
+    [Arguments]    ${path}
+    Open With JupyterLab Menu    File    Open from Path…
+    Input Text    xpath=//input[@placeholder="/path/relative/to/jlab/root"]    ${path}
+    Click Element    xpath://div[.="Open"]
+    Sleep    2s
+    Maybe Close Popup
+    Sleep    2s
+
+Clone Git Repository In Current Folder
```

Comment:
there are already implemented kw to clone a git repository. is there any reasons to create a new one?

Reply:
is that a problem? I see you're also using `Clone Git Repository And Open`
maybe you can change folder before calling the kw

Reply:
[`Clone Git Repository And Open`](https://github.com/red-hat-data-services/ods-ci/blob/c57e8377ece0d3f6e083851262e1cb8adaf0a474/tests/Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot#L247) internally calls `Clone Git Repository`

Reply:
ok..but is it a problem to clone inside /opt/app-root/src?

Reply:
I'm responding for Pranav: yes it is an issue in this case, he needs to clone in a specific folder in order two have two copies of the same repo at different points of their history